tests/apps/core/test_core_api_room_user_access.py

```diff
@@ -12,147 +13,341 @@
+        user = UserFactory()
+        group = GroupFactory(members=[user])
+        jwt_token = AccessToken.for_user(user)
+
+        public_room = RoomFactory(
+            is_public=True, users=[(user, "member")], groups=[(group, "member")]
         )
+        RoomUserAccessFactory(room=public_room)
+        RoomUserAccessFactory(room=public_room, role="member")
+        RoomUserAccessFactory(room=public_room, role="administrator")
+        RoomUserAccessFactory(room=public_room, role="owner")
 
-        self.assertEqual(response.status_code, 401)
-        self.assertEqual(
-            response.json(), {"detail": "Authentication credentials were not provided."}
+        private_room = RoomFactory(is_public=False, users=[(user, "member")])
+        RoomUserAccessFactory(room=private_room)
+        RoomUserAccessFactory(room=private_room, role="member")
+        RoomUserAccessFactory(room=private_room, role="administrator")
+        RoomUserAccessFactory(room=private_room, role="owner")
+
+        response = self.client.get(
+            "/api/room-user-accesses/",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
         )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), [])
 
-    def test_api_room_user_accesses_retrieve_authenticated(self):
+    def test_api_room_user_accesses_list_authenticated_administrator_direct(self):
         """
-        Authenticated users should not be allowed to retrieve a room user access
+        Authenticated users should be allowed to list room user accesses for a room
+        in which they are a direct administrator.
         """
-        access = RoomUserAccessFactory()
-
         user = UserFactory()
         jwt_token = AccessToken.for_user(user)
 
+        public_room = RoomFactory(is_public=True, users=[(user, "administrator")])
+        RoomUserAccessFactory(room=public_room)
+        RoomUserAccessFactory(room=public_room, role="member")
+        RoomUserAccessFactory(room=public_room, role="administrator")
+        RoomUserAccessFactory(room=public_room, role="owner")
+
+        private_room = RoomFactory(is_public=False, users=[(user, "administrator")])
+        RoomUserAccessFactory(room=private_room)
+        RoomUserAccessFactory(room=private_room, role="member")
+        RoomUserAccessFactory(room=private_room, role="administrator")
+        RoomUserAccessFactory(room=private_room, role="owner")
+
         response = self.client.get(
-            f"/api/rooms/{access.room.id!s}/users/{access.user.id!s}/",
+            "/api/room-user-accesses/",
             HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
         )
-        self.assertEqual(response.status_code, 403)
-
-        self.assertEqual(
-            response.json(),
-            {"detail": "You do not have permission to perform this action."},
-        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 10)
```

Comment:
ditto as group file, the 10 results are not clear to me compared to the 8 of the test after, maybe we could do some control on the results returned